File: Tech Module/Programming Fundamentals/18. Dictionaries Lambda and LINQ - Exercises/Dictionaries Lambda and LINQ - Exercises/Phonebook/Phonebook.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Phonebook
{
    class Phonebook
    {
        static Dictionary<string, string> phonebook = new Dictionary<string, string> { };

        static void AddToPhonebook(string name, string number)
        {
            if (phonebook.ContainsKey(name))
            {
                phonebook[name] = number;
            }
            else
            {
                phonebook.Add(name, number);
            }
        }

        static void SearchInPhonebook(string name)
        {
            if (phonebook.ContainsKey(name))
            {
                Console.WriteLine($"{name} -> {phonebook[name]}");
            }
            else
            {
                Console.WriteLine($"Contact {name} does not exist.");
            }
        }

        static void Main(string[] args)
        {
            string command = "";
            while (command != "END")
            {
                string[] input = Console.ReadLine().Split(' ').ToArray();

                command = input[0];
                if (command == "A")
                {
                    string name = input[1];
                    string number = input[2];
                    AddToPhonebook(name, number);

                }
                else if (command == "S")
                {
                    string name = input[1];
                    SearchInPhonebook(name);
                }
            }
        }
    }
}

File: Tech Module/Software Technologies/20. Java Basics - Exercises/src/Phonebook.java
import java.util.HashMap;
import java.util.Locale;
import java.util.Scanner;

import static com.sun.javafx.fxml.expression.Expression.add;

public class Phonebook {
    private static HashMap<String, String> phonebook = new HashMap<>();
    private static Scanner scanner = new Scanner(System.in);
    public static void main(String[] args) {

        while (TakeCommands()) {
        }
    }

    private static boolean TakeCommands() {
        String[] command = scanner.nextLine().split(" ");

        switch (command[0]) {
            case "A":
                String name = command[1];
                String number = command[2];
                add(name, number);
                break;
            case "S":
                name = command[1];
                search(name);
                break;
            case "END":
                return false;
        }
        return true;
    }

    private static void add(String name, String number) {
        phonebook.put(name, number);
    }

    private static void search(String name) {
        if (phonebook.containsKey(name)) {
            System.out.printf("%s -> %s%n", name, phonebook.get(name));
        } else {
            System.out.printf("Contact %s does not exist.%n", name);
        }
    }

}